test/TestProjects/Models-TypeSpec/src/Generated/Models/NumericValuesFacetint32.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Azure.Core;
+
+namespace ModelsTypeSpec.Models
+{
+    /// <summary> The NumericValuesFacetint32. </summary>
+    public partial class NumericValuesFacetint32 : Facet
```

Comment:
I think `NumericValuesFacetint32` is better.
1. It is consistent with Paged model.
2. If spell fails, it is even better because user would know it is not recommended usage.

Reply:
the user only knows it is not recommended when we say it is recommended. In this case, they will just blame us and open an issue.
We should not use mechanism to "imply" a thing is not recommended. For instance, we do not recommend them to use the unions, we generate them as `object`, hide them and leave a message.

Paged model is not a good example here because they usually are internal.